exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryResources.java

```diff
@@ -96,10 +101,13 @@ public QueryResult submitQueryJSON(QueryWrapper query) throws Exception {
   @Produces(MediaType.TEXT_HTML)
   public Viewable submitQuery(@FormParam("query") String query,
                               @FormParam("queryType") String queryType,
-                              @FormParam("autoLimit") String autoLimit) throws Exception {
+                              @FormParam("autoLimit") String autoLimit,
+                              Form form) throws Exception {
```

Comment:
Ah, so you do have a solution for session options. Suggestion: since the two changes affect exactly the same code, just go ahead and combine them. You can create a roll-up JIRA that has the two specific JIRA tickets as sub-tasks.

Reply:
Hmm I originally had them all together and pulled them apart for separate feedback & tracking.  Oops ... maybe.  I personally do prefer small PRs that add specific functionality, though.

Reply:
You have the right idea in general. If we could approve and commit PRs quickly, then smaller would absolutely be better. Unfortunately, it typically takes a week (sometimes several) to get reviews done.

For historical reasons, commits are done only after running tests on a non-public cluster. The folks who have access to that cluster run tests only about once a week, typically on Fridays. To reduce work, that person does a "batch commit", combines that week's commits into a single branch so tests can be run once. It ain't pretty, but it is who things have been done for a long time.

If your code conflicts with another PR (such as your own other change in the same place), the committer will pick one of them, ask you to resolve conflicts, and will then commit the other in the next commit batch a week or two later.

Given our (again, not pretty) process, you are better off combining conflicting changes into a single PR.

This is, by the way, why some of my PRs are big: I can't wait months to get it done in small pieces the way I "should" do it -- I'd completely forget what I was doing several 2-3 week cycles.

We absolutely want to change it, but don't have a good way to run the "pre-commit" tests except on that non-public cluster.

Any suggestions you've seen work on other projects?

Reply:
Well, seems like you've covered it - the tests should be run more often so problems can be picked up faster and merges done whenever people are ready.